staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -43,14 +43,17 @@ func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{},
 
 func nestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, bool, error) {
 	var val interface{} = obj
+	evaluatedFields := make([]string, len(fields))
+
 	for _, field := range fields {
+		evaluatedFields = append(evaluatedFields, field)
```

Comment:
I am worried that this causes too many allocations. This error handling must move into the error code path.

Reply:
e.g. count in the loop and then take the prefix of `fields` of that length.